server/server.go

```diff
@@ -365,6 +369,17 @@ func (s *Server) Kill(connectionID uint64, query bool) {
 	}
```

Comment:
maybe this should be:

```
	if !query {
		// Mark the client connection status as WaitShutdown, when the goroutine detect
		// this, it will end the dispatch loop and exit.
		atomic.StoreInt32(&conn.status, connStatusWaitShutdown)
	}

	conn.mu.RLock()
	cancelFunc := conn.mu.cancelFunc
	conn.mu.RUnlock()
	if cancelFunc != nil {
		cancelFunc()
	}
```

Reply:
done.